bundles/org.openhab.binding.ws980wifi/src/main/java/org/openhab/binding/ws980wifi/internal/discovery/WS980WiFi.java

```diff
@@ -0,0 +1,286 @@
+            this.rainLastYear = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 61, 65)))) / 10;
+            this.rainTotal = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 66, 70)))) / 10;
+            this.lightLevel = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 71, 75)))) / 10;
+            this.uvRaw = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 76, 78))));
+            this.uvIndex = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 79, 80))));
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private byte[] tcpipRequest(byte[] sendBytes) {
+        byte[] receive = {};
+        try (Socket aSocket = new Socket(wsHost, wsPort);
+                OutputStream outStream = aSocket.getOutputStream();
+                DataOutputStream mesgToServer = new DataOutputStream(outStream);
+                InputStream inStream = aSocket.getInputStream();
+                DataInputStream answerFromServer = new DataInputStream(inStream);) {
+            mesgToServer.write(sendBytes, 0, sendBytes.length);
+            receive = new byte[1024];
+            answerFromServer.read(receive);
+            return receive;
+        } catch (IOException e) {
+            log.debug("---> tcpipRequest created an exception: {}", e.getMessage());
+            return receive;
+        }
+    }
+
+    /**
+     * Class Method to discover objects of the class
+     */
+
+    public static WS980WiFi[] discoverDevices(int timeout, InetAddress sourceAddress) {
+        final Logger logger = LoggerFactory.getLogger(WS980WiFi.class);
+        byte[] receBytes = new byte[DISCOVERY_RECEIVE_BUFFER_SIZE];
+        List<WS980WiFi> devices = new ArrayList<WS980WiFi>(50);
+        WS980WiFi[] out = new WS980WiFi[0];
+
+        logger.debug("Discovering WS980WIFI devices");
+
+        try (DatagramSocket sock = new DatagramSocket()) {
+            sock.setBroadcast(true);
+            sock.setReuseAddress(true);
+
+            byte[] localHost = sourceAddress.getAddress();
+            localHost[3] = (byte) 0xff;
+            InetAddress broadcastAddress = InetAddress.getByAddress(localHost);
+
+            byte[] sendBytes = new byte[] { (byte) 0xff, (byte) 0xff, (byte) 0x12, (byte) 0x00, (byte) 0x04,
+                    (byte) 0x16 };
+            DatagramPacket dpSend = new DatagramPacket(sendBytes, sendBytes.length, broadcastAddress,
+                    DISCOVERY_DEST_PORT);
+            sock.send(dpSend);
+            logger.debug("Broadcast has been sent to {} and port {}", broadcastAddress.toString(), DISCOVERY_DEST_PORT);
+
+            DatagramPacket dpReceive = new DatagramPacket(receBytes, receBytes.length);
+            if (timeout == 0) {
+                logger.debug("No discovery timeout was set. Blocking thread until answer received");
```

Comment:
Here and most, if not all, of the remaining logging in this method: Please use **trace** level.